### DISCUSSION TITLE
Container Image - v2.2.0

### DISCUSSION THREAD

**Discussion by tomkerkhove** [Azure/api-management-self-hosted-gateway#205] [Announcements]
## Getting started

You can easily install the self-hosted gateway with Docker:

```shell
docker run -d -p 8080:8080 -p 8081:8081 --name <gateway-name> --envconfig.service.endpoint=<instance-name>.configuration.azure-api.net
config.service.auth=<auth-token> mcr.microsoft.com/azure-api-management/gateway:2.2.0
```

Learn how you can install it on other container platforms:
- [Deploy self-hosted gateway on Kubernetes with Helm](https://aka.ms/apim/sputnik/deploy/k8s/helm)
- [Deploy self-hosted gateway on Kubernetes with Azure Arc (Preview)](https://aka.ms/apim/sputnik/deploy/k8s/arc)
- [Deploy self-hosted gateway on Kubernetes with YAML](https://aka.ms/apim/sputnik/deploy/k8s/yaml)

Here are other relevant resources:
- [Self-hosted gateway on Microsoft Artifact Registry](https://aka.ms/apim/shgw/registry-portal)
- [Migrate to self-hosted gateway v2](https://aka.ms/apim/shgw/migration/v2)
- [Our image tagging strategy](https://aka.ms/apim/shgw/image-tagging)

### Features

- Support for WebSocket APIs
- Support for `validate-azure-ad-token` policy
- Support for `limit-concurrency` policy
- Support for policy fragments
- Support for `context.Deployment.Gateway.*` context properties ([docs](https://learn.microsoft.com/en-us/azure/api-management/api-management-policy-expressions#ContextVariables))

### Fixes / Changes

- Provide a warning when gateway authentication tokens are about to expire ('GatewayTokenAboutToExpire' event)
- Provide additional insights on instance discovery for synchronization
- Provide startup validation to fail fast when misconfigured with better guidance
- Provide overview of supported policies on startup

### Breaking Changes

- Fixed a bug where self-hosted gateway was not respecting API revisions properly
- Fixed a bug where send-one-way-request policy does not gracefully sending a response due to an empty stack
- Fixed a bug where self-hosted gateway could log exceptions up to 6MB in size
- Fixed a bug where local logging was causing a memory leak
- Fixed various vulnerabilities in our base image

### Removal

None.

<hr /><em>This discussion was created from the release <a href='https://github.com/Azure/api-management-self-hosted-gateway/releases/tag/Container-v2.2.0'>Container Image - v2.2.0</a>.</em>

**Comment by oayras** [comment]
Hi @tomkerkhove , is this version preview or stable? The description say "This is a preview release for which ...."

**Reply by tomkerkhove** [reply]
It definately is, but I forgot to update the description - sorry
